src/lib/es2015.reflect.d.ts

```diff
@@ -1,15 +1,43 @@
 declare namespace Reflect {
-    function apply(target: Function, thisArgument: any, argumentsList: ArrayLike<any>): any;
-    function construct(target: Function, argumentsList: ArrayLike<any>, newTarget?: any): any;
+    /**
+     * Calls the function with the specified object as the this value and the elements of specified array as the arguments.
+     * @param target The function to call.
+     * @param thisArgument The object to be used as the this object.
+     * @param argumentsList An array of argument values to be passed to the function.
+     */
+    function apply<T, A extends any[], R>(target: (this: T, ...args: A) => R, thisArgument: T, argumentsList: A): R;
+    function apply(target: (...args: any) => any, thisArgument: any, argumentsList: ArrayLike<any>): any;

```

Comment:
This second overload will just be picked by the compiler whenever anything fails to be accepted by the first overload, so this doesn't really add strictness. Better autocomplete, I guess.

Reply:
Unfortunately, this&nbsp;is&nbsp;needed to&nbsp;be&nbsp;able to&nbsp;pass the&nbsp;`arguments`&nbsp;object.